### PR TITLE
intel_debug: add info for Intel firmware and tplg filenames

### PR DESCRIPTION
**Pull request by kv2019i**
Add tables documenting the firmare and topology filenames and their
locations, as expected by Linux SOF driver.

Related to ongoing discussions in:
 - https://github.com/thesofproject/linux/issues/4590
 - https://github.com/thesofproject/sof/pull/8210

### CODE REVIEW

File: getting_started/intel_debug/introduction.rst [hunk 1]
```diff
@@ -117,13 +117,64 @@ Linux kernel to query whether or not the firmware authentication is enabled,
...
+   :header-rows: 1
+
+   * - Platform
+     - Topology load path
+     - Notes
+   * - Raptor Lake and older
+     - /lib/firmware/intel/sof-tplg/sof-MACH.tplg
+     - MACH = machine driver
+   * - Meteor Lake and newer
+     - /lib/firmware/intel/sof-ace-ipc/sof-MACH.tplg
```
Comment (ujfalusi) [inline getting_started/intel_debug/introduction.rst:171]: /lib/firmware/intel/sof-ace-**tplg**/sof-MACH.tplg

Comment (plbossart) [inline getting_started/intel_debug/introduction.rst:171]: sof-ace-tplg

File: getting_started/intel_debug/introduction.rst [hunk 2]
```diff
@@ -117,13 +117,64 @@ Linux kernel to query whether or not the firmware authentication is enabled,
...
+     - MACH = machine driver
```
Comment (plbossart) [inline getting_started/intel_debug/introduction.rst:172]: not really, this is an indication but not the name of the machine driver, we have e.g. multiple permutations of the hardware supported by the same machine driver but different topology files.

**Review by plbossart — COMMENTED** [review]
one typo and one confusing statement @kv2019i 

All good otherwise for me.

File: getting_started/intel_debug/introduction.rst (hunk 2 above)
Comment (kv2019i) [reply getting_started/intel_debug/introduction.rst:172]: @plbossart true, but couldn't come up with better. maybe I'll put something very generic like CONFIG.

Comment (kv2019i) [reply getting_started/intel_debug/introduction.rst]: modified to CONFIG

File: getting_started/intel_debug/introduction.rst (hunk 1 above)
Comment (kv2019i) [reply getting_started/intel_debug/introduction.rst]: Fixed

**Review by plbossart — DISMISSED** [review]

**Review by ujfalusi — APPROVED** [review]

File: getting_started/intel_debug/introduction.rst [hunk 3]
```diff
@@ -117,13 +117,64 @@ Linux kernel to query whether or not the firmware authentication is enabled,
 which means `dmesg` logs cannot be provided to alert the user to an ME
 configuration issue.
 
+Linux SOF will look up firmware files at following paths:
```
Comment (deb-intel) [inline getting_started/intel_debug/introduction.rst:120]:
```suggestion
Linux SOF will look up firmware files at the following paths:
```

File: getting_started/intel_debug/introduction.rst [hunk 4]
```diff
@@ -117,13 +117,64 @@ Linux kernel to query whether or not the firmware authentication is enabled,
...
+     - PLAT = mtl, lnl, ...
+   * - Meteor Lake and newer Loadable Module
+     - /lib/firmware/intel/sof-ipc4-lib/PLAT/UUID.bin
+     - PLAT as above, UUID = UUID of the module
+   * - Meteor Lake and newer Loadable Module (community signed)
+     - /lib/firmware/intel/sof-ipc4-lib/PLAT/community/UUID.bin
+     - PLAT as above, UUID = UUID of the module
+
+Important notices:
+ - Standard Linux firmware search path and order is followed. The above table covers the base "/lib/firmware" case. Please see https://docs.kernel.org/driver-api/firmware/fw_search_path.html for more information.
```
Comment (deb-intel) [inline getting_started/intel_debug/introduction.rst:149]:
```suggestion
 - The standard Linux firmware search path and order is followed. The above table covers the base "/lib/firmware" case. See https://docs.kernel.org/driver-api/firmware/fw_search_path.html for more information.
```

File: getting_started/intel_debug/introduction.rst [hunk 5]
```diff
@@ -117,13 +117,64 @@ Linux kernel to query whether or not the firmware authentication is enabled,
...
+     - Notes
+   * - Raptor Lake and older
+     - /lib/firmware/intel/sof-tplg/sof-CONFIG.tplg
+     - CONFIG = topology variant needed for detected hardware configuration
+   * - Meteor Lake and newer
+     - /lib/firmware/intel/sof-ace-tplg/sof-CONFIG.tplg
+     - CONFIG = topology variant needed for detected hardware configuration
+
+Important notices:
+ - Standard Linux firmware search path and order is followed. The above table covers the base "/lib/firmware" case. Please see https://docs.kernel.org/driver-api/firmware/fw_search_path.html for more information.
```
Comment (deb-intel) [inline getting_started/intel_debug/introduction.rst:175]:
```suggestion
 - The standard Linux firmware search path and order is followed. The above table covers the base "/lib/firmware" case. See https://docs.kernel.org/driver-api/firmware/fw_search_path.html for more information.
```

**Review by deb-intel — DISMISSED** [review]
Just a few minor grammatical improvements to incorporate.

File: getting_started/intel_debug/introduction.rst [hunk 6]
```diff
@@ -117,13 +117,64 @@ Linux kernel to query whether or not the firmware authentication is enabled,
...
+ - The topology folder and filename can be overridden with "tplg_path" and "tplg_filename" SOF kernel parameters.
```
Comment (marc-hb) [inline getting_started/intel_debug/introduction.rst:176]:
```suggestion
 - The topology folder and filename can be overridden with "tplg_path" and "tplg_filename" `snd_sof_pci` kernel parameters.
```

These could use some documentation too. I could never make any sense of these:

```shell
modinfo snd_sof_pci

parm:           fw_path:alternate path for SOF firmware. (charp)
parm:           fw_filename:alternate filename for SOF firmware. (charp)
parm:           lib_path:alternate path for SOF firmware libraries. (charp)
parm:           tplg_path:alternate path for SOF topology. (charp)
parm:           tplg_filename:alternate filename for SOF topology. (charp)
parm:           sof_pci_debug:SOF PCI debug options (0x0 all off) (int)
```

Can the `_path` include a filename or do you need to specific both? No idea.

**Review by marc-hb — COMMENTED** [review]
Thanks!

**Review by mengdonglin — APPROVED** [review]

**Comment by kv2019i** [comment]
Thanks for reviews. @deb-intel and @marc-hb I merged your wording suggestions to a new version.

File: getting_started/intel_debug/introduction.rst (hunk 6 above)
Comment (kv2019i) [reply getting_started/intel_debug/introduction.rst]: Agreed but these are arguably more in the expert user category and normal end-users or developers should not really need to bother with these.

**Review by lgirdwood — APPROVED** [review]

**Review by btian1 — APPROVED** [review]

**Pull request merged**